api/workers/autoscale.go

```diff
@@ -83,18 +83,20 @@ func autoscaleRack() {
 		return
 	}
 
-	// ok to start multiple
+	log.Logf("change=%d", (desired - system.Count))
+
+	// ok to start multiple instances in one pass
 	// when shutting down go one at a time but only if current status is "running"
 	if desired < system.Count {
 		if system.Status == "running" {
 			system.Count--
+		} else {
+			return
 		}
 	} else {
 		system.Count = desired
 	}
 
-	log.Logf("change=%d", (desired - system.Count))
-
 	err = models.Provider().SystemSave(*system)
```

Comment:
Would be nice to remove the return above and just add a guard here to return if there are no changes.

We should also move the log back down here. I'd like to see the actual changes it's about to try to make after all of the adjustments made by the algorithm.